### PR TITLE
win32: Move Eina_Error error code definitions to appropriate library

### PR DESCRIPTION
**Pull request by Coquinho**

### CODE REVIEW

File: src/lib/evas/canvas/evas_main.c [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #define EFL_INPUT_EVENT_PROTECTED
 
+#include "evas_api.h"
```
Comment (JPTIZ) [inline src/lib/evas/canvas/evas_main.c:3]: If `evas_private` is not including it already, we have a problem houston.

Comment (Coquinho) [reply src/lib/evas/canvas/evas_main.c:3]: I will have a lock to see if we can remove this line.

Comment (Coquinho) [reply src/lib/evas/canvas/evas_main.c:3]: [yup](https://github.com/expertisesolutions/efl/runs/864992329), I'll remove this.

**Comment by Coquinho** [comment]
Sooo, Linux CI is [failing](https://travis-ci.com/github/expertisesolutions/efl/jobs/360430315#L497) with:
```
meson.build:627:8: ERROR: File efl_gfx_error.c does not exist.
```
any ideas?

**Comment by joaoantoniocardoso** [comment]
> Sooo, Linux CI is [failing](https://travis-ci.com/github/expertisesolutions/efl/jobs/360430315#L497) with:
> 
> ```
> meson.build:627:8: ERROR: File efl_gfx_error.c does not exist.
> ```
> 
> any ideas?

It seems to be expecting to be generated from `src/lib/efl/interfaces/efl_gfx_error.eo`, that doesn't exist at all.

**Comment by Coquinho** [comment]
> > Sooo, Linux CI is [failing](https://travis-ci.com/github/expertisesolutions/efl/jobs/360430315#L497) with:
> > ```
> > meson.build:627:8: ERROR: File efl_gfx_error.c does not exist.
> > ```
> > 
> > 
> > any ideas?
> 
> It seems like it is expecting to be generated from `src/lib/efl/interfaces/efl_gfx_error.eo`, that doesn't exists at all. Although this is not the case.
> 
> If we are moving symbols that was defined in evas to efl, I am thinking that maybe evas can't see these symbols anymore, maybe we need to make them visible to evas.

How could we?

**Comment by joaoantoniocardoso** [comment]
> > > Sooo, Linux CI is [failing](https://travis-ci.com/github/expertisesolutions/efl/jobs/360430315#L497) with:
> > > ```
> > > meson.build:627:8: ERROR: File efl_gfx_error.c does not exist.
> > > ```
> > > 
> > > 
> > > any ideas?
> > 
> > 
> > It seems like it is expecting to be generated from `src/lib/efl/interfaces/efl_gfx_error.eo`, that doesn't exists at all. Although this is not the case.
> > If we are moving symbols that was defined in evas to efl, I am thinking that maybe evas can't see these symbols anymore, maybe we need to make them visible to evas.
> 
> How could we?

We could have a `src/lib/efl/efl_gfx_error.h` declaring them with api extern thing..?

**Comment by Coquinho** [comment]
> > > > Sooo, Linux CI is [failing](https://travis-ci.com/github/expertisesolutions/efl/jobs/360430315#L497) with:
> > > > ```
> > > > meson.build:627:8: ERROR: File efl_gfx_error.c does not exist.
> > > > ```
> > > > 
> > > > 
> > > > any ideas?
> > > 
> > > 
> > > It seems like it is expecting to be generated from `src/lib/efl/interfaces/efl_gfx_error.eo`, that doesn't exists at all. Although this is not the case.
> > > If we are moving symbols that was defined in evas to efl, I am thinking that maybe evas can't see these symbols anymore, maybe we need to make them visible to evas.
> > 
> > 
> > How could we?
> 
> We could have a `src/lib/efl/efl_gfx_error.h` declaring them with api extern thing..?

hmm, how exactly? I tried making it a header before, but then my symbols weren't being exported for some reason, I will try again to have a better look at the errors. I will use this branch to test for now.

**Comment by joaoantoniocardoso** [comment]
> > We could have a `src/lib/efl/efl_gfx_error.h` declaring them with api extern thing..?
> 
> hmm, how exactly? I tried making it a header before, but then my symbols weren't being exported for some reason, I will try again to have a better look at the errors. I will use this branch to test for now.

I thought something like this:

src/lib/efl/efl_gfx_error.c:
```c
EFL_API Eina_Error EFL_GFX_IMAGE_LOAD_ERROR_NONE = 0;
```

src/lib/efl/efl_gfx_error.h:
```c
EFL_API extern Eina_Error EFL_GFX_IMAGE_LOAD_ERROR_NONE;
```

Sorry but now i realize that we are looking at wrong place. The error actually tells nothing about the linker and the symbol, it is about meson OR the CI itself. Is it possible to reproduce the same error in my machine?

**Comment by Coquinho** [comment]
ok, I got the same [error](https://github.com/expertisesolutions/efl/pull/300/checks?check_run_id=869606224#step:22:628), any idea @joaoantoniocardoso

**Comment by Coquinho** [comment]
> > > We could have a `src/lib/efl/efl_gfx_error.h` declaring them with api extern thing..?
> > 
> > 
> > hmm, how exactly? I tried making it a header before, but then my symbols weren't being exported for some reason, I will try again to have a better look at the errors. I will use this branch to test for now.
> 
> I thought something like this:
> 
> src/lib/efl/efl_gfx_error.c:
> 
> ```c
> EFL_API Eina_Error EFL_GFX_IMAGE_LOAD_ERROR_NONE = 0;
> ```
> 
> src/lib/efl/efl_gfx_error.h:
> 
> ```c
> EFL_API extern Eina_Error EFL_GFX_IMAGE_LOAD_ERROR_NONE;
> ```
> 
> Sorry but now I realize that we are looking at wrong place. The error actually tells nothing about the linker and the symbol, it is about meson OR the CI itself. Is it possible to reproduce the same error in my machine?

I couldn't get it.

**Comment by Coquinho** [comment]
@joaoantoniocardoso I will try as you said.

**Comment by JPTIZ** [comment]
> 
> 
> > > We could have a `src/lib/efl/efl_gfx_error.h` declaring them with api extern thing..?
> > 
> > 
> > hmm, how exactly? I tried making it a header before, but then my symbols weren't being exported for some reason, I will try again to have a better look at the errors. I will use this branch to test for now.
> 
> I thought something like this:
> 
> src/lib/efl/efl_gfx_error.c:
> 
> ```c
> EFL_API Eina_Error EFL_GFX_IMAGE_LOAD_ERROR_NONE = 0;
> ```
> 
> src/lib/efl/efl_gfx_error.h:
> 
> ```c
> EFL_API extern Eina_Error EFL_GFX_IMAGE_LOAD_ERROR_NONE;
> ```
> 
> Sorry but now I realize that we are looking at wrong place. The error actually tells nothing about the linker and the symbol, it is about meson OR the CI itself. Is it possible to reproduce the same error in my machine?

That's what I was going to point out. The problem was never a symbol defined or anything, it is **the file** who is not found.

**Comment by Coquinho** [comment]
> > > > We could have a `src/lib/efl/efl_gfx_error.h` declaring them with api extern thing..?
> > > 
> > > 
> > > hmm, how exactly? I tried making it a header before, but then my symbols weren't being exported for some reason, I will try again to have a better look at the errors. I will use this branch to test for now.
> > 
> > 
> > I thought something like this:
> > src/lib/efl/efl_gfx_error.c:
> > ```c
> > EFL_API Eina_Error EFL_GFX_IMAGE_LOAD_ERROR_NONE = 0;
> > ```
> > 
> > 
> > src/lib/efl/efl_gfx_error.h:
> > ```c
> > EFL_API extern Eina_Error EFL_GFX_IMAGE_LOAD_ERROR_NONE;
> > ```
> > 
> > 
> > Sorry but now I realize that we are looking at wrong place. The error actually tells nothing about the linker and the symbol, it is about meson OR the CI itself. Is it possible to reproduce the same error in my machine?
> 
> That's what I was going to point out. The problem was never a symbol defined or anything, it is **the file** who is not found.

Yeah, I know, but I don't have any better idea. Do you have one?

**Comment by JPTIZ** [comment]
> Yeah, I know, but I don't have any better idea. Do you have one?

Get a working branch and see what happens to that file. When is it added, etc.

**Comment by joaoantoniocardoso** [comment]
How could I reproduce the CI error?
```
meson.build:630:8: ERROR: File efl_gfx_error.c does not exist.
```
I just tried to delete the file to see the error message in the case of the file *really* doesn't exist and here you are:
```
src/lib/efl/meson.build:22:0: ERROR: File efl_gfx_error.c does not exist.
```

File: src/lib/evil/evil_time.h [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 #include <evil_api.h>
 
+#include <check.h>
```
Comment (JPTIZ) [inline src/lib/evil/evil_time.h:6]: Wait what

Comment (JPTIZ) [reply src/lib/evil/evil_time.h:6]: This shouldn't be here. At all. Check is for test-units, why is it here?

Comment (joaoantoniocardoso) [reply src/lib/evil/evil_time.h:6]: hahaha he is just testing the issue of having both gettimeofday

Comment (Coquinho) [reply src/lib/evil/evil_time.h:6]: oh no wrong branch

Comment (JPTIZ) [reply src/lib/evil/evil_time.h:6]: >:(

Comment (JPTIZ) [reply src/lib/evil/evil_time.h:6]: Damn, markdown turned my angry-face into a sad face

**Comment by Coquinho** [comment]
> > Yeah, I know, but I don't have any better idea. Do you have one?
> 
> Get a working branch and see what happens to that file. When is it added, etc.

This was add in native-windows first, It gets the same error. (will rebase this to native windows, don't know why I've rebased it with another branch)

**Comment by Coquinho** [comment]
> > Yeah, I know, but I don't have any better idea. Do you have one?
> 
> Get a working branch and see what happens to that file. When is it added, etc.

okay, so if I understand right, I've done as you said, @JPTIZ, so we could see the error.

File: src/lib/efl/meson.build [hunk 3]
```diff
@@ -8,7 +8,7 @@ efl_header_src = [
   'Efl_MVVM_Common.h'
 ]
 
-efl_src = []
+efl_src = ['efl_gfx_error.c']
```
Comment (JPTIZ) [inline src/lib/efl/meson.build]:
```suggestion
efl_src = files('efl_gfx_error.c')
```

This should get the reference to the file properly.

**Pull request merged**